src/compositions/sidebar.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { FC, useEffect, useState } from 'react';
+import { HeatingIcon } from '../icons/fire';
+import { UsersIcon } from '../icons/usersIcon';
+import { Page } from '../layouts/page';
+import { Heading2 } from '../identity/heading-2';
+import { LightningIcon } from '../icons/lightningIcon';
+import { PaperAirplaneIcon } from '../icons/paperAirplaneIcon';
+import { LinkElement } from '../elements/link';
+import { Question } from '../data/questions';
+import { useRouter } from 'next/dist/client/router';
+import { SaveAsIcon } from '@heroicons/react/outline';
+
+const categoryNavigation = [
+  { name: 'Team', icon: UsersIcon },
+  { name: 'Strom', icon: LightningIcon },
+  { name: 'Heizung', icon: HeatingIcon },
+  { name: 'Transport', icon: PaperAirplaneIcon },
+  { name: 'Reisen', icon: PaperAirplaneIcon },
+];
+
+type Props = { question?: Question; categoriesWithIndexes?: { [key: string]: string[] } };
+type NavigationProps = { className?: string };
+type PropsSaveDataElement = { onClick?: () => void };
+
+const styleKategories = 'group flex items-center md:justify-start text-xs px-2 py-2 md:text-sm font-medium rounded-md';
+
+const SaveDataElement: FC<PropsSaveDataElement> = ({ onClick }) => {
+  return (
+    <div className="group flex-shrink-0 flex md:border-t md:border-gray-200 md:p-4 md:w-full md:group md:block ">
+      <LinkElement href="/rechner/saved" onClick={onClick}>
+        <SaveAsIcon className="md:hidden h-6 w-6 justify-center items-center m-2"></SaveAsIcon>
+        <div className="hidden md:contents">
+          <p className="text-sm font-medium text-gray-700 group-hover:text-gray-900 flex justify-start">Take a break?</p>
+          <p className="text-xs font-medium text-gray-500 group-hover:text-gray-900 flex justify-start">
+            Save your answers here
+          </p>
+        </div>
+      </LinkElement>
+    </div>
+  );
+};
+
+export const CategoriesNavigation: FC<Props> = ({ children, question, categoriesWithIndexes }) => {
+  const [categoryChangedManually, setCategoryChangedManually] = useState('0');
+  const [currentCategory, setCurrentCategorie] = useState(
+    categoryNavigation.find((c) => c.name === question?.category) || categoryNavigation[0]
+  );
+  const router = useRouter();
+  useEffect(() => {
+    setCurrentCategorie(categoryNavigation.find((c) => c.name === question?.category) || categoryNavigation[0]);
+  }, [router]);
+
+  useEffect(() => {
+    router.push(`/rechner/${categoryChangedManually || '1'}`);
+  }, [categoryChangedManually]);
```

Comment:
I don't really understand why you created a new State (`categoryChangedManually `) which gets set `onClick` in the navigation below to trigger a side effect which routes to the relevant page?
Why not directly route to the relevant page instead of setting some state in `onClick` in the navigation? Or even better use a real link instead of navigating with javascript...